music21/romanText/writeRoman.py

```diff
@@ -0,0 +1,481 @@
+    >>> rnWriterFromMeasure.combinedList == rnWriterFromRn.combinedList
+    True
+
+    >>> p = stream.Part()
+    >>> p.insert(0, m)
+    >>> rnWriterFromPart = romanText.writeRoman.RnWriter(p)
+    >>> rnWriterFromPart.combinedList == rnWriterFromMeasure.combinedList
+    True
+
+    >>> s = stream.Score()
+    >>> s.insert(0, m)
+    >>> rnWriterFromScoreWithoutPart = romanText.writeRoman.RnWriter(p)
+    >>> rnWriterFromScoreWithoutPart.combinedList == rnWriterFromMeasure.combinedList
+    True
+
+    In all cases, handling of composer and work metadata is
+    extracted from score metadata wherever possible.
+    A metadata.composer entry will register directly as will any enties for
+    workTitle, movementNumber, and movementName (see the prepTitle method for details).
+    As always, these entries are user-settable.
+    Make any adjustments to the metadata before calling this class.
+    '''
+
+    def __init__(self,
+                 obj: base.Music21Object,
+                 ):
+
+        self.composer = 'Composer unknown'
+        self.title = 'Title unknown'
+
+        if obj.isStream:
+
+            if isinstance(obj, stream.Score):
+                if obj.parts:
+                    self.container = obj.parts[0]
+                else:  # score with no parts
+                    self.container = obj
+
+            elif isinstance(obj, stream.Part):
+                self.container = obj
+
+            elif isinstance(obj, stream.Measure):
+                self.container = stream.Part()
+                self.container.insert(0, obj)
+
+            else:  # A stream, but not a measure, part, or score
+                objs = [x for x in obj]
+                self._makeContainer(objs)
+
+            if obj.metadata:  # sic, obj not container for metadata, and only if it's stream
+                self.prepTitle(obj.metadata)
+                if obj.metadata.composer:
+                    self.composer = obj.metadata.composer
+
+        else:  # Not a stream
+            self._makeContainer([obj])
+
+        self.combinedList = [f'Composer: {self.composer}',
+                             f'Title: {self.title}',
+                             'Analyst: ',
+                             'Proofreader: ',
+                             '']  # One blank line between metadata and analysis
+        # Note: blank analyst and proof reader entries until supported within music21 metadata
+
+        if not self.container.getElementsByClass('TimeSignature'):
+            self.container.insert(0, meter.TimeSignature('4/4'))  # Placeholder
+
+        self.currentKeyString: str = ''
+        self.prepSequentialListOfLines()
+
+    def _makeContainer(self,
+                       objs: list):
+        '''
+        Makes a placeholder container for the usual cases where this class is called on
+        generic- or non-stream object.
+        '''
+        m = stream.Measure()
+        for x in objs:
+            m.insert(0, x)
```

Comment:
This will prepend the element each time, essentially reversing the order. What about just `.append()`?

Reply:
Thanks @jacobtylerwalls . 

By potentially brittle tests, yes I meant the 0 and -1 cases – glad you agree that they should be fine. 

The next commit includes opus, without separating off a new function. Grateful for your views on this solution.

Thanks again.